crates/rrg/src/fs.rs

```diff
@@ -135,13 +88,39 @@ pub fn list_dir<P: AsRef<Path>>(path: P) -> std::io::Result<ListDir> {
 ///
 /// The iterator can be constructed with the [`walk_dir`] function.
 pub struct WalkDir {
+    max_depth: u32,
     iter: ListDir,
     pending_iters: Vec<std::io::Result<ListDir>>,
     #[cfg(target_family = "unix")] dev: u64,
 }
 
 impl WalkDir {
 
+    /// Limits recursion to the specified `max_depth`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the given limit is zero.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// let paths = rrg::fs::walk_dir("/").unwrap().with_max_depth(1)
+    ///     .filter_map(Result::ok)
+    ///     .map(|entry| entry.path)
+    ///     .collect::<Vec<_>>();
+    ///
+    /// assert!(paths.contains(&"/usr".into()));
+    /// assert!(!paths.contains(&"/usr/bin".into()));
+    /// assert!(!paths.contains(&"/usr/lib".into()));
+    /// ```
+    pub fn with_max_depth(mut self, max_depth: u32) -> WalkDir {
+        assert!(max_depth > 0);
+
+        self.max_depth = max_depth;
+        self
+    }
+
     #[cfg(target_family = "unix")]
```

Comment:
What happens on macos?

Reply:
`unix` option of [`target_family`] covers both macOS and Linux, as opposed to [`target_os`] that disambiguates between the two.

[`target_family`]: https://doc.rust-lang.org/reference/conditional-compilation.html#target_family
[`target_os`]: https://doc.rust-lang.org/reference/conditional-compilation.html#target_os